src/java/org/apache/cassandra/db/memtable/TrieMemtable.java

```diff
@@ -369,16 +370,36 @@ public FlushablePartitionSet<MemtablePartition> getFlushSet(PartitionPosition fr
         long partitionKeySize;
         long partitionCount;
 
-        if (allPositionsToFlush)
+        int fromShard = Integer.MIN_VALUE;
```

Comment:
This should work.

However, it could be a bit simpler, and perhaps more efficient, if we use `getShardForToken` for both boundaries. If from/to is a `maxKeyBound` and the returned shard's start boundary matches its token, then the span fully includes (resp. excludes) that shard.

E.g. if `getShardForToken(from) == 2` and `getShardForToken(to) == 4`, and the tokens match, add the precollected counts/sizes for shards 2 and 3.
If `getShardForToken(from) == 2` and `getShardForToken(to) == 4`, and neither token matches, walk shard 2 from `from` and shard 4 to `to` and add the precollected count and size for shard 3.